lib/src/resources/movie_repository.dart

```diff
@@ -0,0 +1,9 @@
+import 'dart:async';
+import 'movie_api_provider.dart';
+import '../models/movie_model.dart';
+
+class MovieRepository {
```

Comment:
This repository should also be implementing an interface

Reply:
Done